### DISCUSSION TITLE
mock fails to create icingadb from fedora-sources: "go: command not found"

### DISCUSSION THREAD

**Discussion by vpdose** [rpm-software-management/mock#1056] [General]
### Short description of the problem

I am trying to create CentOS Stream packages for Icinga2. I am using the src-rpm from [packages.icinga.com/fedora/](https://packages.icinga.com/fedora/37/release/src/

Almost every package can be created, mock is really great, your work is very much appreciated!!

But I am failing to create the icingadb package. I do believe, there is something missing - golang, as far I as understand the process , because the error message is "go: command not found". If someone could guide me in the right direction where to look ... maybe I need to change some setting in the src-file???

This is the output from mock:

Finish: build setup for icingadb-1.1.0-1.fc37.src.rpm
Start: Outputting list of installed packages
Finish: Outputting list of installed packages
Start: rpmbuild icingadb-1.1.0-1.fc37.src.rpm
Building target platforms: x86_64
Building for target x86_64
Executing(%prep): /bin/sh -e /var/tmp/rpm-tmp.SjFnBJ
+ umask 022
+ cd /builddir/build/BUILD
+ cd /builddir/build/BUILD
+ rm -rf icingadb-1.1.0
+ /usr/bin/gzip -dc /builddir/build/SOURCES/icingadb-1.1.0.tgz
+ /usr/bin/tar -xof -
+ STATUS=0
+ '[' 0 -ne 0 ']'
+ cd icingadb-1.1.0
+ /usr/bin/chmod -Rf a+rX,u+w,g-w,o-w .
+ go build -buildvcs=false -trimpath -o . ./cmd/...
/var/tmp/rpm-tmp.SjFnBJ: line 48: **go: command not found**
error: Bad exit status from /var/tmp/rpm-tmp.SjFnBJ (%prep)

RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.SjFnBJ (%prep)
Finish: rpmbuild icingadb-1.1.0-1.fc37.src.rpm
Finish: build phase for icingadb-1.1.0-1.fc37.src.rpm
ERROR: Exception(/tmp/icingadb-1.1.0-1.fc37.src.rpm) Config(centos-stream+epel-8-x86_64) 0 minutes 37 seconds
INFO: Results and/or logs in: /var/lib/mock/centos-stream+epel-8-x86_64/result
ERROR: Command failed:

 /usr/bin/systemd-nspawn -q -M 9d748204142b47638d1d714b14b3fab1 -D /var/lib/mock/centos-stream+epel-8-x86_64/root -a -u mockbuild --capability=cap_ipc_lock --bind=/tmp/mock-resolv.m6vikrrc:/etc/resolv.conf --bind=/dev/mapper/control --bind=/dev/loop-control --bind=/dev/loop0 --bind=/dev/loop1 --bind=/dev/loop2 --bind=/dev/loop3 --bind=/dev/loop4 --bind=/dev/loop5 --bind=/dev/loop6 --bind=/dev/loop7 --bind=/dev/loop8 --bind=/dev/loop9 --bind=/dev/loop10 --bind=/dev/loop11 --setenv=TERM=vt100 --setenv=SHELL=/bin/bash --setenv=HOME=/builddir --setenv=HOSTNAME=mock --setenv=PATH=/usr/bin:/bin:/usr/sbin:/sbin --setenv=PROMPT_COMMAND=printf "\033]0;<mock-chroot>\007" --setenv=PS1=<mock-chroot> \s-\v\$  --setenv=LANG=C.UTF-8 --resolv-conf=off bash --login -c /usr/bin/rpmbuild -bb --noclean --target x86_64 --nodeps /builddir/build/SPECS/icingadb.spec

### Output of `rpm -q mock`

[root@SERVER ~]# rpm -q mock
mock-3.5-1.el8.noarch

### Steps to reproduce issue

1. Download the fedora package from here: https://packages.icinga.com/fedora/37/release/src/icingadb/icingadb-1.1.0-1.fc37.src.rpm
2.  mock -r centos-stream+epel-8-x86_64 /tmp/icingadb-1.1.0-1.fc37.src.rpm

Do not forget to mention full commandline with the mock command you executed.

**Comment by xsuchy** [comment]
Mock install in the buildroot only the minimal set of packages. And go is not in the minimal set. So it has to be specified as  build dependency.
The command go is provided by:

    $ rpm -qf /usr/bin/go
    golang-bin-1.20.3-1.fc38.x86_64

The spec file of [icingadb-1.1.0-1.fc37.src.rpm](https://packages.icinga.com/fedora/37/release/src/icingadb/icingadb-1.1.0-1.fc37.src.rpm) is missing it. You have to add to the spec file this line:

    BuildRequires: golang-bin

**Reply by vpdose** [reply]
Hi xsuchy,

thank you very much for this tip. 

I am still failing to compile icingadb, because go needs to download a lot of libraries and this is not allowed in the chroot.

Even after adding 

config_opts['rpmbuild_networking']

the process seems not to be able to download stuff.

I now compiled and build the rpm without mock and go a package.

Thanks a lot !

Best regards
Volker
